ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -755,14 +756,17 @@ static MM_BOOLEAN MM_UpdateEntireHeader(struct MM_DATA_BASE_XP *data_base_XP)
     /* from 4 a 7, position MM_FIRST_OFFSET_to_N_RECORDS */
     if (data_base_XP->nRecords > UINT32_MAX)
     {
-        if (fwrite_function(&data_base_XP->nRecords, 4, 1,
+        GUInt32 nRecords32HightBits = (GUInt32)(data_base_XP->nRecords >> 32);
+        if (fwrite_function(&nRecords32HightBits, 4, 1,
```

Comment:
No that's wrong!
The original code write the 32 bit *low* bits in this field. So you've changed its behaviour. Actually you could use the below else  branch you've modified in both cases. You always want to write the low 32 bits of data_base_XP->nRecords

Reply:
proposed fix in https://github.com/AbelPau/gdal/pull/13

Reply:
Thanks!